wdl2cwl/main.py

```diff
@@ -36,784 +21,788 @@
+                dockerpull_expr, WDL.Expr.Ident
+            ):
+                raise Exception(
+                    f"Unsupported type: {type(dockerpull_expr)}: {dockerpull_expr}"
+                )
+            dockerpull_referee = dockerpull_expr.referee
+            if dockerpull_referee is None:
+                raise Exception(f"Unsupported type: {type(dockerpull_referee)}")
+            dockerpull = dockerpull_referee.expr.literal.value
+        return cwl.DockerRequirement(dockerPull=dockerpull)
+
+    def get_cwl_command_requirements(
+        self, wdl_commands: List[Union[str, WDL.Expr.Placeholder]]
+    ) -> cwl.InitialWorkDirRequirement:
+        """Translate WDL commands into CWL Initial WorkDir REquirement."""
+        command_str: str = ""
+        for wdl_command in wdl_commands:
+            if isinstance(wdl_command, str):
+                command_str += wdl_command.replace("$(", "\\$(")
+            elif isinstance(wdl_command, WDL.Expr.Placeholder):
+                command_str += self.translate_wdl_placeholder(wdl_command)
+
+        command_str = textwrap.dedent(command_str)
+        return cwl.InitialWorkDirRequirement(
+            listing=[cwl.Dirent(entry=command_str, entryname="example.sh")]
         )
-    )
 
-    requirements.append(cwl.InlineJavascriptRequirement())
-    requirements.append(cwl.NetworkAccess(networkAccess=True))
-
-    if "memory" in ast.task_runtime:
-
-        ram_min: Union[str, int] = ""
-
-        if '"' in ast.task_runtime["memory"]:
-            if "${" in ast.task_runtime["memory"]:
-                input_name = ast.task_runtime["memory"][
-                    ast.task_runtime["memory"].find("${")
-                    + 2 : ast.task_runtime["memory"].find("}")
-                ]
-                temp = ast.task_runtime["memory"].index("}")  # index of }
-                if len(ast.task_runtime["memory"]) != temp + 1:
-                    unit = ast.task_runtime["memory"][temp + 1 : -1].strip()
-                    if input_name in input_names:
-                        ram_min = get_ram_min_js(input_name, unit)
+    def translate_wdl_placeholder(self, wdl_placeholder: WDL.Expr.Placeholder) -> str:
+        """Translate WDL Expr Placeholder to a valid CWL command string."""
+        cwl_command_str = ""
+        expr = wdl_placeholder.expr
+        if expr is None:
+            raise Exception(f"Placeholder '{wdl_placeholder}' has no expr.")
+        placeholder_expr = self.get_expr(expr)
+        options = wdl_placeholder.options
+        if options:
+            if "true" in options:
+                true_value = options["true"]
+                false_value = options["false"]
+                if type(true_value) == str:
+                    true_str = (
+                        f'"{true_value}"'
+                        if '"' not in true_value
+                        else f"'{true_value}'"
+                    )
+                else:
+                    true_str = true_value
```

Comment:
This line is not called. Can you create a copy of an existing WDL file and modify it so that this code path is tested?

Reply:
I just confirmed from the wdl spec that both true and false always have a string literal. I will remove the condition check